### PR TITLE
Add support for glob tests

### PR DESCRIPTION
**Pull request by CAD97**
This patch adds basic support for performing snapshot testing based on files matching a glob and transformed by some operation.

### CODE REVIEW

**Comment by CAD97** [comment]
(Local failure of glob tests was a line ending issue, along with my computer probably being "helpful".)

**Comment by CAD97** [comment]
This is a bit messy, but it works! It could use some attention to the usage API, but I'm not familiar enough with all the variations of how you can call insta to really flesh out the macros here.

The snapshots for glob tests are placed in a directory structure directly mirroring that the input test files came from.

Whether the globs should be relative to the test file or the test working directory should be considered; I did the working directory here, but relative to the test file also has a strong argument, I'm just not sure the best way to do it.

**Comment by mitsuhiko** [comment]
I really like the idea but I think we might want to store the "input" that feeds into the execution into the meta data that is persisted in the file. Code generally became quite messy recently unfortunately :(

File: tests/snapshots/test_glob__identity_glob_example/test_data/example.txt.snap [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+source: tests/test_glob.rs
+expression: "std::convert::identity"
```
Comment (mitsuhiko) [inline tests/snapshots/test_glob__identity_glob_example/test_data/example.txt.snap:3]: So ideally we have something like `input: example.txt` here.

Comment (CAD97) [reply tests/snapshots/test_glob__identity_glob_example/test_data/example.txt.snap:3]: I definitely agree but I have no idea where/how to add that metadata :sweat_smile:

**Comment by CAD97** [comment]
Glob test snapshot files' metadata now include the input path.

No separator normalization occurs, so windows puts `\\` and I assume nixes will put `/`. The input metadata not matching seems to not matter to anything.

**Comment by CAD97** [comment]
And speaking of normalization, I tried and have no idea why it didn't work...

```rust
Some(dbg!(dbg!(path_str).replace(r"\\", "/"))),
```

```
---- identity_glob_example stdout ----
[src\runtime.rs:1060] path_str = "test_data\\example copy.txt"
[src\runtime.rs:1060] dbg!(path_str).replace(r"\\", "/") = "test_data\\example copy.txt"
[src\runtime.rs:1060] path_str = "test_data\\example.txt"
[src\runtime.rs:1060] dbg!(path_str).replace(r"\\", "/") = "test_data\\example.txt"
```

**Comment by mitsuhiko** [comment]
Closing this for #112. For more information see that ticket.

**Comment by CAD97** [comment]
Nice solution!

**Pull request closed without merging**